posthog/test/base.py

```diff
@@ -557,10 +557,30 @@ def execute_wrapper(query, *args, **kwargs):
             yield queries
 
 
+class RaiseOnJoinThread(threading.Thread):
+    """
+    By default, the thread will swallow exceptions and not raise them in the
+    main thread. This class will raise them in the main thread such that we do
+    not fail silently e.g. on failures to apply ClickHouse schemas.
+    """
+
+    def run(self):
+        self.exc = None
+        try:
+            super().run()
+        except BaseException as e:
+            self.exc = e
+
+    def join(self):
+        super().join()
+        if self.exc:
+            raise self.exc
```

Comment:
This is not passing a `timeout` argument to the `super`'s `join`, so the API is not compatible.